token/core/zkatdlog/nogh/sender.go

```diff
@@ -51,30 +60,36 @@ func (s *Service) Transfer(txID string, wallet driver.OwnerWallet, ids []*token3
 		owners = append(owners, output.Owner.Raw)
 		ownerIdentities = append(ownerIdentities, output.Owner.Raw)
 	}
+	// produce zkatdlog transfer action
+	// return for each output its information in the clear
 	transfer, infos, err := sender.GenerateZKTransfer(values, owners)
 	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed generating zkat proof for txid [%s]", txID)
+		return nil, nil, errors.Wrapf(err, "failed to generate zkatdlog transfer action for txid [%s]", txID)
 	}
 
-	// Prepare metadata
-	infoRaws := [][]byte{}
+	// prepare metadata
+	var infoRaws [][]byte
 	for _, information := range infos {
 		raw, err := information.Serialize()
 		if err != nil {
-			return nil, nil, errors.WithMessage(err, "failed serializing token info")
+			return nil, nil, errors.WithMessage(err, "failed serializing token info for zkatdlog transfer action")
 		}
 		infoRaws = append(infoRaws, raw)
 	}
-
+	// audit info for receivers
 	var receiverAuditInfos [][]byte
-	for _, output := range outputTokens {
+	for i, output := range outputTokens {
+		if output.Owner == nil {
```

Comment:
and what if this is a redeem?

Reply:
even when it is redeem, owner is initialized.